python/adjoint/objective.py

```diff
@@ -483,3 +495,123 @@ def __call__(self):
+        The biggest warning there is to be careful to avoid aliasing if
+        the fields vary quickly in time.
+        Note on yee_grid: For the Poynting Flux to work, H and E components
+        must lie at the same points. Therefore, the Yee grid will always be false.
+    """
+    def __init__(
+        self,
+        sim,
+        volume,
+        scale = 1,
+        decimation_factor=0
+    ):
+        super().__init__(sim)
+        #_fit_volume_to_simulation increases the dimensionality of
+        #the volume, so we'll use the user input volume
+        self.volume = sim._fit_volume_to_simulation(volume)
+        self.decimation_factor = decimation_factor
+        self.scale = scale
+        #get_normal returns an index for the two 
+        # dictionaries of cross products
+        self.normal = self.get_normal(volume)
+
+    def register_monitors(self, frequencies):
+        self._frequencies = np.asarray(frequencies)
+
+        #Add the dft monitors for the interesting components
+        self._monitor =self.sim.add_dft_fields(EH_TRANSVERSE[self.normal], 
+                                            frequencies, 
+                                            where = self.volume,
+                                            yee_grid = False) 
+
+        return self._monitor
+
+    def place_adjoint_source(self, dJ):
+        dJ = np.atleast_1d(dJ)
+        if dJ.ndim == 2:
+            dJ = np.sum(dJ, axis=1)
+        time_src = self._create_time_profile()
+        scale = self._adj_src_scale()
+        if self._frequencies.size == 1:
+            amp =  dJ * scale
+            src = time_src
+        else:
+            scale =  dJ * scale
+            src = FilteredSource(
+                time_src.frequency,
+                self._frequencies,
+                scale,
+                self.sim.fields.dt,
+            )
+            amp = 1
+        source = []
+        #TODO: account for sign in the current sources
+        #(the K sources should have the negative)
+        for field_t,field in  enumerate(JK_TRANSVERSE[self.normal]):
+            #dft_fields returns a scalar value for components that aren't evaluated
+            #in these cases, we don't need to add an adjoint source
+            if(self.field_component_evaluations[field_t].size != 1):
+                tuple_elements = np.zeros((1,self.metadata[3].size,1), dtype=np.complex128)
+                #TODO: Load up the source data for other normal vectors
+                #TODO: Remove this unnecessary for loop
+                for index,element in enumerate(self.field_component_evaluations[field_t]):
+                    tuple_elements[0,index,0] = element
+                source.append( mp.Source(
+                    src,
+                    component = field,
+                    amplitude=amp,
+                    size=self.volume.size,
+                    center=self.volume.center,
+                    amp_data = tuple_elements
```

Comment:
You need to conjugate some of the fields, right?

Reply:
Some of the current sources have a sign change too. Is that implemented elsewhere?

Reply:
These are resolved as of the most recent update